### ISSUE TITLE
[Bug] ME Export Bus Issue with IC2 Coolant Cells and Fuzzy Card

### ISSUE THREAD

**Issue by gatoborrachon** [PrototypeTrousers/Applied-Energistics-2#19]
The ME Export Bus does not recognize any 'IC2 Coolant Cell' with less than 99% percent of durability when using a Fuzzy Card with the configuration "Ignore"

**Describe the bug**
When i tried to insert IC2 Coolant Cells into an IC2 Reactor, the ME Export Bus does not insert any IC2 Coolant Cell with less than 99% of life (anything below the full life (for example: 60000) is not recognized, for example: 59999/60000)

**To Reproduce**
1.- Add IC2 and AE2

2.- Place an IC2 Reactor with some Fuel Cells, place an Export Bus on the Left and an Import Bus on the Right, each Bus will have its own ME Chest, just like this:
![2021-05-18_09 47 31](https://user-images.githubusercontent.com/61126174/118674131-2c9c2800-b7bf-11eb-9a15-65e364dc798c.png)

3.- You will require to damage some Coolant Cells, use 10k coolant cells (to speed up the process), any kind of Nuclear Fuel (Uranium or MOX, it doesn't matter) and turn on the reactor. Example (here i'm using MOX Fuel Cells and 10k coolant cells, the reactor plating is optional): 
![2021-05-18_09 57 23](https://user-images.githubusercontent.com/61126174/118674530-7553e100-b7bf-11eb-90dc-b503c37bb521.png)

4.- add 1 fuzzy card and 3 speed cards on each Bus (Importer and Exporter)

5.- Before the Coolant Cells are too damaged, take one and put it inside the ME Export Bus with the comparison: Ignore.
![2021-05-18_09 47 46](https://user-images.githubusercontent.com/61126174/118675300-1b9fe680-b7c0-11eb-885f-8a655f4012e4.png)

5.1.- Duplicate an entire stack of these Slightly Damaged Coolant Cells and put it inside the ME Chest of the Export Bus Side.
![2021-05-18_10 06 27](https://user-images.githubusercontent.com/61126174/118676119-c0babf00-b7c0-11eb-8bea-f4b4f3fdf56f.png)

6.- Once you damaged enough your Coolant Cells (for example: 2500/10000) you can use one of them as a template for the Import Bus (set the comparison to 25%). Example (notice the damaged Coolant Cell):
![2021-05-18_09 47 54](https://user-images.githubusercontent.com/61126174/118675053-eb584800-b7bf-11eb-9388-3c61a0a88203.png)

7.- You will see how the Damaged Coolant Cells are being removed, but they are not being replenished even when the ME Chest of the Exporter has enough Coolant Cells.
(Reactor Empty)
![2021-05-18_10 06 25](https://user-images.githubusercontent.com/61126174/118676098-bd273800-b7c0-11eb-9e28-2cf281fd22f0.png)

(ME Chest full)
![2021-05-18_10 06 27](https://user-images.githubusercontent.com/61126174/118676119-c0babf00-b7c0-11eb-8bea-f4b4f3fdf56f.png)

**Expected behavior**
In short, the ME Export Bus should insert these Slightly damaged Coolant Cells, but it simply doesn't.

**Additional context**
If you ask why i have Slightly Damaged Coolant Cells, i use another Reactor where i regen the Coolant Cells, and they doesn't fill with an exact value (around 56%)

**Environment**
+There are no more requeriments
+No more Mods are required.

- Minecraft Version: 1.12.2
- AE2 Version: appliedenergistics2-rv6-stable-7.omni-fixes-v40c
- Forge Version: 1.12.2-forge-14.23.5.2854
- IC2 Version: industrialcraft-2-2.8.221-ex112

**Comment by Exaxxion** [comment]
I suspect (but haven't looked at it beyond your screenshots) that this is the case because the fuzzy damage function compares items which store damage in a vanilla way using metadata values. I think the coolant cells are storing their damage using NBT tags (which is interpreted by AE2 as them being distinct items).

**Comment by gatoborrachon** [comment]
well, i think you are right:

1.- i have 2 IC2 addons (GregTech Experimental and FrogCraft) that add Coolant Cells

2.- GregTech Experimental (and IC2) Coolant Cells store their damage using the NBT "AdvDmg"
2.1.- But FrogCraft uses Metadata, and in one test i did, it seemed to work (i will do a new try to confirm this)

3.- But as far as i know, someone in the Unofficial AE2 Discord Server told me that the Fuzzy Card recognizes NBT too, but idk if it only recognize a NBT called 'damage' or something like this, so i think that we just need to add that NBT 'AdvDmg' as one of the recognized NBT values (if this is the case of course).

**Comment by gatoborrachon** [comment]
sorry, i can't believe i forgot it

> I think the coolant cells are storing their damage using NBT tags (which is interpreted by AE2 as them being distinct items).

but then why the Import Bus works perfectly fine? (maybe inside the code of the Import bus there's the key to solve this issue, but i really don't know)

**Comment by PrototypeTrousers** [comment]
heres an old issue on the original ae2 issue-tracker explaining why it this way:
https://github.com/AppliedEnergistics/Applied-Energistics-2/issues/3361

TL;DR: I couldnt make it work.

**Comment by gatoborrachon** [comment]
> TL;DR: I couldnt make it work.

we can't ask for help from that guy? (the guy from this video: https://www.youtube.com/watch?v=VB0lgmBu0pI)

EDIT: He said this:
> I have already deleted that code, but tomorrow I can look again and describe in more detail how to fix it

**Comment by theArhar** [comment]
So I searched my computer and found some difference between github version of rv5-stable.5-1.12 and what I have now
I am adding files mentioned in [previous topic](https://github.com/AppliedEnergistics/Applied-Energistics-2/issues/3361#issuecomment-361575577)
[new.zip](https://github.com/PrototypeTrousers/Applied-Energistics-2/files/6524275/new.zip)
in ItemComparisonHelper I added block from line 106
in AEItemStack - getMaxDamage 338
in AESharedItemStack - many alters, line 47, isDamageable, getCustomDamage, getMaxDamage, etc, see comparison
also it can be that github version is slightly different from what I have downloaded 3 years ago, so feel free to ask more questions

**Comment by gatoborrachon** [comment]
> in AEItemStack - getMaxDamage 338

mmm, well, it seems that many things changed since rv5-stable.5-1.12, i simply can't override `getMaxDamage` 

> in AESharedItemStack - many alters, line 47, isDamageable, getCustomDamage, getMaxDamage, etc, see comparison

here too,  no one of these functions are able to be overrided

**Comment by theArhar** [comment]
How do you trying to compile your project? What IDE do you use?

**Comment by theArhar** [comment]
I managed to successfully rewrite rv6-stable7 code with same fix (some places are different, but logic the same) and it works
![2021-05-22_14 22 26](https://user-images.githubusercontent.com/8017191/119224917-7cedef80-bb09-11eb-911e-f4391025c516.png)
![2021-05-22_14 22 46](https://user-images.githubusercontent.com/8017191/119224921-7e1f1c80-bb09-11eb-8530-14d348fb9425.png)
![2021-05-22_14 23 04](https://user-images.githubusercontent.com/8017191/119224922-7eb7b300-bb09-11eb-83c2-f16ae12b9433.png)
![2021-05-22_14 23 09](https://user-images.githubusercontent.com/8017191/119224924-7eb7b300-bb09-11eb-9037-e2c0bdeb4f26.png)
![2021-05-22_14 23 18](https://user-images.githubusercontent.com/8017191/119224925-7eb7b300-bb09-11eb-9965-e3749d02ec86.png)

**Comment by theArhar** [comment]
notice 2499 damage on pic.2 - this is crucial
edited files will be in zip
[rv6-stable7.zip](https://github.com/PrototypeTrousers/Applied-Energistics-2/files/6526303/rv6-stable7.zip)

**Comment by theArhar** [comment]
notice, you probably want to remove AELog (125,126,127) in ItemComparisonHelper

**Comment by gatoborrachon** [comment]
my god, thank you so much, i will build the dev environment with the fixes ASAP

> How do you trying to compile your project? What IDE do you use?

As always, `gradlew setupDecompWorkspace`, and since i'm using Eclipse: `gradlew eclipse`

**Comment by gatoborrachon** [comment]
sorry for the relay, i were a bit busy

and yeah, it totally worked, every coolant cell is being moved at the specified damage, i also tried the ME System with the Regen Reactor and it works precfectly too, many thanks

System working
![2021-05-22_17 52 55](https://user-images.githubusercontent.com/61126174/119242826-d31d5b80-bb26-11eb-943f-d2fa58d62a46.png)


System with the Regen Reactor
![2021-05-22_17 52 58](https://user-images.githubusercontent.com/61126174/119242850-052ebd80-bb27-11eb-8e07-e97cbfbf37c6.png)


Cells being inserted:
![2021-05-22_17 53 08](https://user-images.githubusercontent.com/61126174/119242864-2099c880-bb27-11eb-93b6-8dc9f1b3f6c5.png)


Cells being removed (50% in Fuzzy Card):
![2021-05-22_17 56 39](https://user-images.githubusercontent.com/61126174/119242867-2394b900-bb27-11eb-8a53-c25c084fb524.png)
